algorithms/src/snark/varuna/data_structures/proof.rs

```diff
@@ -378,6 +380,88 @@ impl<E: PairingEngine> FromBytes for Proof<E> {
+    let n_instances: usize = batch_sizes.iter().sum();
+
+    match varuna_version {
+        VarunaVersion::V1 => None,
+        VarunaVersion::V2 => {
+            // All fields are serialised in Compressed mode The breakdown is as
+            // follows:
+            // - batch sizes: (boils down to CanonicalSerialize for [usize])
+            //   + one u64 for the length (the number of circuits) followed that many usize
+            //     This contains the size information for the vectors in all other fields,
+            //     which are therefore serialised without their length
+            // - commitments:
+            //   + witness_commitments: n_instances commitments
+            //   + mask_poly: 1 byte to encode the enum tag (a bool) plus one commitment if
+            //     the variant is Some (if and only if the proof system is run in ZK mode)
+            //   + h_0, g_1, g_2, h_2: four commitments
+            //   + g_a, g_b, g_c: 3 * n_circuits commitments
+            // - evaluations:
+            //   + g_1_eval: one field element
+            //   + g_a_evals, g_b_evals, g_c_evals: 3 * n_circuits field elements
+            // - third_msg:
+            //   + 3 * n_instances field elements
+            // - fourth_msg:
+            //   + 3 * n_circuits field elements
+            // - pc_proof:
+            //   + one usize for the size of the vector (which is always 3)
+            //   + three of [1 commitment + 1 bool (for the Option tag) + {1 field element
+            //     if the variant is Some (if and only if the proof system is run in ZK
+            //     mode)}]
+
+            let n_bool = 1;
+            let n_u64 = 1;
+            let n_field_elements = 1 + 6 * n_circuits + 3 * n_instances;
+            let n_commitments = 4 + n_instances + (if hiding { 1 } else { 0 }) + 3 * n_circuits;
+
+            // The next three sizes are const functions
+            let size_bool = size_of::<bool>();
+            let size_u64 = size_of::<u64>();
+            // The next two can be hard-coded if performance becomes critical
+            // and they are considered fully stable. They are 32 and 48 bytes at
+            // the time of writing, respectively (commitments are affine points
+            // written in compressed form).
+            let size_field_element = E::Fr::one().compressed_size();
+            let size_commitment = KZGCommitment::<E>::empty().compressed_size();
+
+            let size_pc_proof =
+                size_of::<usize>() + 3 * (size_commitment + 1) + if hiding { size_field_element } else { 0 };
+
+            Some(
+                n_bool * size_bool
+                    + n_u64 * size_u64
+                    + size_of_val(batch_sizes)
+                    + n_field_elements * size_field_element
+                    + n_commitments * size_commitment
+                    + size_pc_proof,
+            )
+
+            // NB: "n_usize * size_usize" must be computed as
```

Comment:
```suggestion
            // Note. "n_usize * size_usize" must be computed as
```
Minor nit. Had to look up "NB" to know what this meant :)

Reply:
Ah, sorry! I usually write it like that without even thinking :) Will pay more attention when documenting in the future. Fortunately, the comment is gone because there are no longer usize computations in the function ^^